modern/src/runtime/AnimatedLayer.ts

```diff
@@ -2,6 +2,97 @@ import Layer from "./Layer";
 import { unimplementedWarning } from "../utils";
 
 class AnimatedLayer extends Layer {
+  _playing: boolean;
+  _frameNum: number;
+
+  constructor(node, parent, annotations, store) {
+    super(node, parent, annotations, store);
+
+    if (!node.attributes.hasOwnProperty("autoplay")) {
+      node.attributes.autoplay = "0";
+    }
+
+    if (!node.attributes.hasOwnProperty("autoreplay")) {
+      node.attributes.autoreplay = "1";
+    }
+
+    if (!node.attributes.hasOwnProperty("speed")) {
+      node.attributes.speed = "200";
+    }
+
+    node.attributes.autoplay = !!Number(node.attributes.autoplay);
+    node.attributes.autoreplay = !!Number(node.attributes.autoreplay);
+
+    this._initializeStartEnd(node);
+
+    this._playing = node.attributes.autoplay;
+    this._frameNum = node.attributes.start || 0;
+
+    this.js_listen("js_framechange", () => {
+      this._frameNum += 1;
+      if (this._frameNum > Number(this.getendframe())) {
+        this._frameNum = this.getstartframe();
+        if (!node.attributes.autoreplay) {
+          return;
+        }
+      }
+      this.js_trigger("js_update");
+
+      if (this._playing) {
+        setTimeout(
+          () => this.js_trigger("js_framechange"),
+          Number(node.attributes.speed)
+        );
+      }
+    });
+
+    if (this._playing) {
+      this.js_trigger("js_framechange");
+    }
+  }
+
+  _initializeStartEnd(node) {
```

Comment:
Could we call this `_initializeStartAndEnd`. On first reading, I thought it initialized the beginning _of_ the end of the animation which was super confusing 😛

Reply:
=X for sure